subsys/net/ip/tcp.c

```diff
@@ -661,12 +661,47 @@ int net_tcp_prepare_ack(struct net_tcp *tcp, const struct sockaddr *remote,
 	return -EINVAL;
 }
 
+static void get_src_addr_ptr(struct net_tcp *tcp,
+			     const struct sockaddr *local,
+			     struct sockaddr_ptr *addr)
+{
+	/* If the source address is not provided then consider
+	 * context local address is the source address.
+	 */
+	if (!local) {
+		goto end;
+	}
+
+	memset(addr, 0, sizeof(struct sockaddr_ptr));
+
+#if defined(CONFIG_NET_IPV4)
+	if (local->sa_family == AF_INET) {
+		net_sin_ptr(addr)->sin_family = AF_INET;
+		net_sin_ptr(addr)->sin_port = net_sin(local)->sin_port;
+		net_sin_ptr(addr)->sin_addr = &net_sin(local)->sin_addr;
+	}
+#endif
+
+#if defined(CONFIG_NET_IPV6)
+	if (local->sa_family == AF_INET6) {
+		net_sin6_ptr(addr)->sin6_family = AF_INET6;
+		net_sin6_ptr(addr)->sin6_port = net_sin6(local)->sin6_port;
+		net_sin6_ptr(addr)->sin6_addr = &net_sin6(local)->sin6_addr;
+	}
+#endif
+
+end:
+	addr = &tcp->context->local;
```

Comment:
This is a dead statement, and apparently doesn't do what was intended by the author (what?).